PostHog/Autocapture/AutocaptureEventProcessing.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  AutocaptureEventProcessing.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 22/10/2024.
+//
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    import UIKit
+
+    protocol AutocaptureEventProcessing: AnyObject {
+        func process(source: PostHogAutocaptureIntegration.EventData.EventSource, event: PostHogAutocaptureIntegration.EventData)
+    }
+
+    class PostHogAutocaptureEventProcessor: AutocaptureEventProcessing {
+        private weak var postHogInstance: PostHogSDK?
+        private weak var postHogConfig: PostHogConfig?
+        private var debounceTimers: [Int: Timer] = [:]
+
+        init(postHogInstance: PostHogSDK, configuration: PostHogConfig) {
+            self.postHogInstance = postHogInstance
+            self.postHogConfig = configuration
+            PostHogAutocaptureIntegration.addEventProcessor(self)
+        }
+
+        deinit {
+            PostHogAutocaptureIntegration.removeEventProcessor(self)
+        }
+
+        /**
+         Processes an autocapture event, with optional debounce logic for controls that emit frequent events.
+
+         - Parameters:
+            - source: The source of the event (e.g., gesture recognizer, action method, or notification).
+            - event: The autocapture event data, containing properties, screen name, and other metadata.
+
+         If the event has a `debounceInterval` greater than 0, the event is debounced.
+         This is useful for UIControls like `UISlider` that emit frequent value changes, ensuring only the last value is captured.
+         The debounce interval is defined per UIControl by the `ph_autocaptureDebounceInterval` property of `AutoCapturable`
+         */
+        func process(source: PostHogAutocaptureIntegration.EventData.EventSource, event: PostHogAutocaptureIntegration.EventData) {
+            assert(Thread.isMainThread, "Event captured off main thread")
```

Comment:
I think this would crash the app and we don't want that.

Reply:
Since user's builds are not in our control, as an SDK, crashing is not acceptable, in this case, I'd be strongly against `assert`s, `hedgeLog`, or any other alternative that would be better here because this will come back and bite us, defensive programming is very important in SDK dev.